extension/test/integration/get-origin-keys.handler.spec.js

```diff
@@ -0,0 +1,119 @@
+    await iTSetUp.initServerAndExtension(ctpClient)
+  })
+
+  after(async () => {
+    await iTSetUp.cleanupResources(ctpClient)
+  })
+
+  it('given a payment ' +
+    'when getOriginKeysRequest custom field is set and custom field getOriginKeysResponse is not set ' +
+    'then should set custom field getOriginKeysResponse ' +
+    'and interface interaction with type getOriginKeysRequest', async () => {
+    const getOriginKeysRequestDraft = {
+      originDomains: [
+        "http://localhost:8080",
+        "http://localhost:8081"
+      ]
+    }
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: c.CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: "type",
+          key: c.CTP_WEB_COMPONENTS_PAYMENT_TYPE_KEY
+        },
+        fields: {
+          getOriginKeysRequest: JSON.stringify(getOriginKeysRequestDraft)
+        }
+      }
+    }
+
+    const { statusCode, body: payment } = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    expect(statusCode).to.equal(201)
+
+    const { getOriginKeysRequest, getOriginKeysResponse } = payment.custom.fields
+    expect(getOriginKeysRequest).to.be.deep.equal(JSON.stringify(getOriginKeysRequestDraft))
+
+    const interfaceInteraction = payment.interfaceInteractions
+      .find(interaction => interaction.fields.type === c.CTP_INTERACTION_TYPE_GET_ORIGIN_KEYS)
+    expect(interfaceInteraction).to.not.undefined
+    expect(getOriginKeysResponse).to.be.deep.equal(interfaceInteraction.fields.response)
+
+    expect(JSON.parse(interfaceInteraction.fields.request)).to.be.deep.equal({
+      merchantAccount: process.env.ADYEN_MERCHANT_ACCOUNT,
+      ...getOriginKeysRequestDraft
+    })
+
+    const interfaceInteractionResponse = JSON.parse(interfaceInteraction.fields.response)
+    expect(interfaceInteractionResponse.originKeys).to.exist
+  })
+
+  it('given a payment ' +
+    'when getOriginKeysRequest is not includes originDomains array and custom field getOriginKeysResponse is not set ' +
+    'then should set custom field with a serialized error getOriginKeysResponse with ' +
+    'and interface interaction with type getOriginKeysRequest', async () => {
+    const getOriginKeysRequestDraft = { } //an invalid request
+
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: c.CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: "type",
+          key: c.CTP_WEB_COMPONENTS_PAYMENT_TYPE_KEY
+        },
+        fields: {
+          getOriginKeysRequest: JSON.stringify(getOriginKeysRequestDraft)
+        }
+      }
+    }
+
+    const { statusCode, body: payment } = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    expect(statusCode).to.equal(201)
```

Comment:
What is the reason that we create in the case of error and not throw an error to the platform? Can you add a comment to the task with explanation? We will later add the reason to our docs. https://jira.commercetools.com/browse/CTPI-209

Reply:
> we create in the case of error and not throw an error to the platform?

I am also not sure from that, What will you throw from the platform? 
Because here in that example, the error is an interaction with Adyen. I guess this should not be thrown,
but when you check the other parts we don't have a proper error handling mechanism.
Which I already add those things as a todo, so I could get some feedback because there are a couple of ways to implement those and could be changed with other parts.

https://github.com/commercetools/commercetools-adyen-integration/blob/00a9cba17a27e41703c928d0eef7e7d35f54e6c0/extension/src/paymentHandler/payment-handler.js#L11-L16

Another point in here, I am also not throwing an error it could be the requestString could be invalid, which could be wrong. So, in summary, we could have lots of edge cases to handle errors and we need to discuss those, which that PR could be the best candidate for this.

https://github.com/commercetools/commercetools-adyen-integration/blob/2fad63252f9fb3da0f6faa80dc9b17fd94b8a13d/extension/src/paymentHandler/web-component-service.js#L15-L23

Reply:
It's ok for me too if we discuss about common error handling in a different task. Create a JIRA so that we don't forget if it's not already created.

Reply:
No, I meant that:
> which that PR could be the best candidate for this.

Reply:
Which PR is that PR?

Reply:
current PR.

Reply:
it seems we have some issues created for error handling: https://github.com/commercetools/commercetools-adyen-integration/issues/46